packages/service-library/src/servicelib/deferred_tasks/_redis_memory_manager.py

```diff
@@ -0,0 +1,52 @@
+from typing import Final
+from uuid import uuid4
+
+from ..redis import RedisClientSDKHealthChecked
+from ..utils import logged_gather
+from ._base_memory_manager import BaseMemoryManager
+from ._models import TaskUID
+from ._task_schedule import TaskSchedule
+
+_MEMORY_MANAGER_PREFIX: Final[str] = "mm:"
+
+
+def _get_key(task_uid: TaskUID) -> str:
+    return f"{_MEMORY_MANAGER_PREFIX}{task_uid}"
+
+
+class RedisMemoryManager(BaseMemoryManager):
+    def __init__(self, redis_sdk: RedisClientSDKHealthChecked) -> None:
+        self.redis_sdk = redis_sdk
+
+    async def get_task_unique_identifier(self) -> TaskUID:
+        candidate_already_exists = True
+        while candidate_already_exists:
+            candidate = f"{uuid4()}"
+            candidate_already_exists = (
+                await self.redis_sdk.redis.get(_get_key(candidate)) is not None
+            )
+        return TaskUID(candidate)
+
+    async def _get_raw(self, redis_key: str) -> TaskSchedule | None:
+        found_data = await self.redis_sdk.redis.get(redis_key)
+        return None if found_data is None else TaskSchedule.parse_raw(found_data)
```

Comment:
is this heavy? cause using pydantic for data that you fully control is not very reasonable performance wise